api/provider/aws/test_aws/environment_delete_test.go

```diff
@@ -117,3 +119,50 @@ func TestDeleteEnvironmentIdempotence(t *testing.T) {
 		t.Fatal(err)
 	}
 }
+
+func TestCheckEnvironmentDependencies(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	mockAWS := awsc.NewMockClient(ctrl)
+	tagStore := tag.NewMemoryStore()
+	mockConfig := mock_config.NewMockAPIConfig(ctrl)
+
+	mockConfig.EXPECT().Instance().Return("test").AnyTimes()
+
+	envTag := models.Tag{
```

Comment:
[nit] even though it's a little less efficient, it may read better to keep the tag insertions together, e.g.:
```go
for _, c := range cases {
    tags := []models.Tag{
        { environment tag here },
        { entity tag here },
    }

    for _, tag := range tags {
        if err := tagStore.Insert(tag); err != nil { ... }
    }

    ...
}
```